src/telemetry/traces.ts

```diff
@@ -6,6 +6,31 @@ import { TelemetryCollector } from './collector';
 import { hrTime } from '@opentelemetry/core';
 import { globalParams } from '../utils';
 
+import { context, trace } from '@opentelemetry/api';
+import { AsyncLocalStorageContextManager } from '@opentelemetry/context-async-hooks';
+
+export function isTraceContextWorking(): boolean {
+  const span = trace.getTracer('otel-bootstrap-check').startSpan('probe');
+  const testContext = trace.setSpan(context.active(), span);
+
+  let visible: boolean | undefined;
+  context.with(testContext, () => {
+    visible = trace.getSpan(context.active()) === span;
+  });
+
+  span.end?.();
+  return visible === true;
+}
+
+export function installTraceContextManager() {
+  const contextManager = new AsyncLocalStorageContextManager();
+  contextManager.enable();
+  context.setGlobalContextManager(contextManager);
+
+  const provider = new BasicTracerProvider();
+  provider.register();
```

Comment:
AFAIK If we do this before starting the Tracer, the registration we do in the Tracer constructor will be a no-op and the resource will be lost

```
      resource: new Resource({
        [SemanticResourceAttributes.SERVICE_NAME]: 'dbos',
      }),
```